src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -5943,16 +5943,7 @@ void ThreadSuspend::SuspendEE(SUSPEND_REASON reason)
             // that there is a thread which appears to be stopped at a gc
             // safe point, but which really is not. If that is the case,
             // back off and try again.
-            //
-            // When the debugger is synchronizing, trying to perform a GC could deadlock. The GC has the
-            // threadstore lock and synchronization cannot complete until the debugger can get the
-            // threadstore lock. However the GC can not complete until it sends the BeforeGarbageCollection
-            // event, and the event can not be sent until the debugger is synchronized. In order to break
-            // this deadlock cycle the GC must give up the threadstore lock, allow the debugger to synchronize,
-            // then try again.
-            // 
-            || (CORDebuggerAttached() &&
-                (g_pDebugInterface->ThreadsAtUnsafePlaces() || g_pDebugInterface->IsSynchronizing()))
+            || (CORDebuggerAttached() && g_pDebugInterface->ThreadsAtUnsafePlaces())
```

Comment:
This is a simplification of @kouvel's change as proposed in  https://github.com/dotnet/runtime/pull/44539#discussion_r521580819